### PR TITLE
Deslop core, delete dead code, fix copy/freeze bugs

### PR DESCRIPTION
**Pull request by aidenybai**
## Summary
A cleanup pass over `packages/react-grab` (plus one CLI nit) that is strictly **net-subtractive** — no new abstraction layers. Deletes verified dead code, removes duplication, and fixes four real bugs surgically.

**Structure (no new concepts):**
- Extracted the selection-label lifecycle (timer map + create/cancel/fade/update) into `core/label-controller.ts`, draining ~115 lines from the `core/index.tsx` god-file, and deduped the copy flow (`buildLabelInstance`, `runLabeledCopy`).
- Descriptive renames (`deltaX`/`deltaY`, `parseHexChannels` channels), a few `as` casts → `instanceof`/type-guards, and DRY constant reuse (`EDIT_LABEL_CLASS`, `EDIT_TRANSPARENT_COLOR_HEX`).

**Dead code removed (all confirmed unused via grep):**
- `store.theme` and `lastCopiedElement` (write-only state) + `clearLastCopied`.
- Dead props/params: `TagBadge.forceShowIcon`, `DropdownAnchor.toolbarWidth` (+ `EDIT_PANEL_POINTER_ANCHOR_WIDTH_PX`), `StepArrow.disabled`/`onPointerUp`/`onPointerLeave`, `ToolbarContent.collapseButton`, `registerOverlayDismiss.onConfirm`, `createElementSelector.shouldUseFinder`.
- Unexported internal-only interfaces; deduped `EDIT_LABEL_CLASS`.

**Bug fixes (surgical, in-place):**
- **Copy success-on-failure** — `copyElementsToClipboard` now returns the real result and `executeCopyOperation` honors it, so a failed clipboard write shows the error label + unfreezes instead of falsely showing "Copied".
- **Label fade-removal timer leak** — the nested removal timeout is now tracked, so it's cancellable on hover/dispose.
- **`setToolbarState`** now honors an explicit `state.enabled` instead of always deriving it from `collapsed`.
- **`selectionLabelVisible`** now respects the `theme.enabled` kill-switch like its sibling memos.

Net: 358 insertions / 353 deletions across 29 files.

## Test plan
- [x] `pnpm typecheck`
- [x] `pnpm lint` (0 warnings, 0 errors)
- [x] `pnpm format`
- [x] `pnpm build`
- [x] `pnpm test` — 650 passing; the lone failure (`edge-cases › recover after target element is removed`) is a pre-existing timing flake (3/3 green in isolation) and touches none of the changed paths.

---

> [!NOTE]
> **Medium Risk**
> Changes touch core activation/copy/state paths and toolbar enablement, but behavior fixes are localized and most edits are removals or extractions without new features.
> 
> **Overview**
> This PR is a **net-subtractive cleanup** of `packages/react-grab` (plus a small CLI tweak) that removes dead code, pulls selection-label logic out of the core entrypoint, and fixes a few interaction bugs.
> 
> **Core / copy flow:** Label create/fade/update logic moves into `core/label-controller.ts`, with shared helpers (`buildLabelInstance`, `runLabeledCopy`) replacing duplicated paths in `core/index.tsx`. Clipboard copy now **returns and respects a boolean** so failed writes show error state and unfreeze instead of treating success as copied. Unused grab-store fields (`theme`, `lastCopiedElement`, `clearLastCopied`) and `completeCopy(element)` are dropped.
> 
> **UI / API trimming:** Removes unused props and anchors (`TagBadge.forceShowIcon`, `DropdownAnchor.toolbarWidth`, toolbar `collapseButton` override, `StepArrow` disabled/pointer-up handlers, overlay dismiss `onConfirm`). Edit panel labels share `EDIT_LABEL_CLASS`. `createElementSelector` always runs the finder path (no `shouldUseFinder` flag).
> 
> **Bug fixes:** Label fade schedules a removal timeout that is **tracked and cancellable** on hover/dispose; hover resume also applies to error/fading labels. `setToolbarState` honors explicit `enabled`; `selectionLabelVisible` gates on global theme enabled like other overlay memos.
> 
> **Misc:** CLI `CHROMIUM_CUSTOM_FORMAT` is module-private; small renames and `instanceof`/channel naming cleanups across utils.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit da0ffff6ca12be8bbb0894462648fce57726ad80. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

---
## Summary by cubic
Cleanup of `packages/react-grab` that deletes dead code, extracts the selection-label lifecycle into `core/label-controller.ts`, and fixes copy/toolbar/label visibility bugs. Copy feedback is now accurate, label timers don’t leak or stick, and toolbar enablement is predictable.

- **Bug Fixes**
  - Copy result is honored and plugin intercept results are awaited even with mixed selections; failed copies show an error and unfreeze instead of “Copied”.
  - Fixed label fade/removal leak by tracking and cancelling nested timers; unhover now reschedules fade for error/fading labels so they don’t stick.
  - `setToolbarState` respects an explicit `enabled` value.
  - Selection label visibility now honors the theme kill-switch.

- **Refactors**
  - Moved label lifecycle into `core/label-controller.ts`; deduped `buildLabelInstance` and labeled copy flow.
  - Removed unused state/props: `store.theme`, `lastCopiedElement`, and stale component props; dropped `EDIT_PANEL_POINTER_ANCHOR_WIDTH_PX`.
  - Safer typing and clearer names: `instanceof` guards, `deltaX/deltaY`, `parseHexChannels` uses `red/green/blue/alpha`, and shared `EDIT_LABEL_CLASS`.
  - Simplified APIs: removed `createElementSelector.shouldUseFinder`, `registerOverlayDismiss.onConfirm`, and the toolbar collapse button override.

<sup>Written for commit da0ffff6ca12be8bbb0894462648fce57726ad80. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/aidenybai/react-grab/pull/444?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: packages/react-grab/src/core/label-controller.ts [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+    scheduleFade(instanceId);
+  };
+
+  const handleHoverChange = (instanceId: string, isHovered: boolean) => {
+    if (isHovered) {
+      cancelFade(instanceId);
+      return;
+    }
+    const instance = getLabelInstances().find((labelInstance) => labelInstance.id === instanceId);
+    if (instance && instance.status === "copied") {
```
[the remark this answers is not in the document]
Comment (aidenybai) [reply packages/react-grab/src/core/label-controller.ts]: Valid — fixed in da0ffff6. `handleHoverChange` now reschedules the fade for `error` and `fading` labels (not just `copied`) on unhover.

File: packages/react-grab/src/core/index.tsx [hunk 2]
```diff
@@ -817,15 +679,45 @@ export const init = (rawOptions?: Options): ReactGrabAPI => {
         }
       }
       await waitUntilNextFrame();
+
+      let didCopy = true;
```
[the remark this answers is not in the document]
Comment (aidenybai) [reply packages/react-grab/src/core/index.tsx:683]: Valid — fixed in da0ffff6. Changed the `else if` to `if` so intercepted plugin `pendingResults` are awaited and validated even when the selection also has unhandled elements.

File: packages/react-grab/src/core/index.tsx [hunk 3]
```diff
@@ -817,15 +679,46 @@ export const init = (rawOptions?: Options): ReactGrabAPI => {
...
+        didCopy = await copyResolvedElements(unhandledElements, extraPrompt, resolvedComponentName);
+      }
+      if (pendingResults.length > 0) {
         const results = await Promise.all(pendingResults);
         if (!results.every(Boolean)) {
           throw new CopyFailedError();
         }
       }
       void notifyElementsSelected(targetElements);
+      return didCopy;
```
[the remark this answers is not in the document]
Comment (aidenybai) [reply packages/react-grab/src/core/index.tsx:694]: Intentional, declining. This is a rare mixed selection (a plugin intercepts *some* elements while the clipboard path handles the rest) where the clipboard write fails. In that case the user's clipboard genuinely does not contain the unhandled elements, so reporting failure (and showing the error label) is the honest, safer outcome — claiming success when the primary clipboard write failed would be misleading. Both failure paths (pending `throw` and unhandled `false`) surface identically as an error label, so observable behavior is consistent. The PR's goal here was narrower — stop a fully-failed copy from showing "Copied" — which is fixed. Strict "fail if any part fails" semantics are deliberate; I'd rather not add partial-success bookkeeping for this edge case.
